chia/util/profiler.py

```diff
@@ -0,0 +1,150 @@
+import asyncio
+import cProfile
+import logging
+import pathlib
+
+from chia.util.path import mkdir, path_from_root
+
+# to use the profiler, enable it config file, "enable_profiler"
+# the output will be printed to your chia root path, e.g. ~/.chia/mainnet/profile/
+# to analyze the profile, run:
+
+#   python chia/utils/profiler.py ~/.chia/mainnet/profile | less -r
+
+# this will print CPU usage of the chia full node main thread at 1 second increments.
+# find a time window of interest and analyze the profile file (which are in pstats format).
+
+# for example:
+
+#   python chia/utils/profiler.py ~/.chia/mainnet/profile 10 20
+
+
+async def profile_task(root_path: pathlib.Path, log: logging.Logger) -> None:
+
+    profile_dir = path_from_root(root_path, "profile")
+    log.info("Starting profiler. saving to %s" % profile_dir)
+    mkdir(profile_dir)
+
+    counter = 0
+
+    while True:
+        pr = cProfile.Profile()
+        pr.enable()
+        # this will throw CancelledError when we're exiting
+        await asyncio.sleep(1)
+        pr.create_stats()
+        pr.dump_stats(profile_dir / ("slot-%05d.profile" % counter))
+        log.debug("saving profile %05d" % counter)
+        counter += 1
+
+
+if __name__ == "__main__":
+    import sys
+    import pstats
+    import io
+    from colorama import init, Fore, Back, Style
+    from subprocess import check_call
+
+    profile_dir = pathlib.Path(sys.argv[1])
```

Comment:
fyi you can avoid the sys.argv list by using `argparse` and add variable names to what you read in.